src/main/java/com/google/javascript/clutz/DeclarationGenerator.java

```diff
@@ -1398,10 +1398,31 @@ private void visitVarDeclaration(String name, JSType type) {
     }
 
     private void visitTemplateTypes(ObjectType type) {
-      visitTemplateTypes(type, Collections.<String>emptySet());
+      visitTemplateTypes(type, Collections.<String>emptySet(), true);
     }
 
-    private void visitTemplateTypes(ObjectType type, Set<String> alreadyEmittedTemplateType) {
+    /**
+     * Emits template types for a given type. For example for
+     *
+     * <pre>T<A,B></pre>
+     *
+     * , this method will emit
```

Comment:
wow, that's a very ugly formatting google-java-format chose there. But maybe it's wrapping `<pre>` because that's a block element, and this would look better if you used `<code>`?

Reply:
much better.